provision/pull-images.sh

```diff
@@ -50,7 +50,7 @@ if [ -z "${NAME_PREFIX}" ]; then
         k8s.gcr.io/coredns:1.7.0 \
         k8s.gcr.io/coredns/coredns:v1.8.0 \
         quay.io/cilium/alpine-curl:v1.3.0 \
-        quay.io/cilium/cilium-builder:e1d9538b22cafeec22068e1483bd5ff51276fadc \
+        quay.io/cilium/cilium-builder:417fc5466e58b71be1d39ed53d1270368bdfc2db \
```

Comment:
I think we can remove this image from the VMs because Cilium images are built using GitHub actions now.

Reply:
The old tag was out of date anyway, I'll update this PR to remove cilium-builder image.

Reply:
But K8sCustomCalls still uses that image: https://github.com/cilium/cilium/blob/master/test/k8sT/manifests/demo-customcalls.yaml#L102.

Reply:
OK, will not change anything then.